include/nuttx/fs/fs.h

```diff
@@ -466,6 +466,7 @@ typedef struct cookie_io_functions_t
 struct file
 {
   int               f_oflags;   /* Open mode flags */
+  int               f_refs;     /* Reference count */
```

Comment:
let us add a kconfig option to support disable this feature to save the code size

Reply:
Will this add a lot of code size？It's not good to add too many restriction macros to code

Reply:
You can consider resolving similar issues in a more elegant way, eg:

```
#ifdef CONFIG_FS_REFCOUNT
int fs_putfilep(FAR struct file *filep)
#else
#  define fs_putfilep(f)
#endif
```

and also, any changes you need to consider the situation of resource-constrained devices. If you think this feature must exist and cannot be restricted by kconfig, then you should choose linux instead of nuttx

Reply:
![20240829-124137](https://github.com/user-attachments/assets/bcc92e67-4eee-449d-b25a-eedf61107e93)

Reply:
I see. Let me think about how to optimize。thanks

Reply:
> You can consider resolving similar issues in a more elegant way, eg:
> 
> ```
> #ifdef CONFIG_FS_REFCOUNT
> int fs_putfilep(FAR struct file *filep)
> #else
> #  define fs_putfilep(f)
> #endif
> ```
> 
> and also, any changes you need to consider the situation of resource-constrained devices. If you think this feature must exist and cannot be restricted by kconfig, then you should choose linux instead of nuttx

but fs_putfilep need free filep if it's the last reference. Without reference count, it' hard to determine whether fs_putfilep should free or not free filep.

Reply:
we could keep the orignal free point with a special name(maybe `fs_freefilep`?), eg: 

```
#ifdef CONFIG_FS_REFCOUNT
int fs_putfilep(FAR struct file *filep);
#  define fs_freefilep fs_putfilep
#else
int fs_freefilep(FAR struct file *filep);
#  define fs_putfilep(f)
#endif
```